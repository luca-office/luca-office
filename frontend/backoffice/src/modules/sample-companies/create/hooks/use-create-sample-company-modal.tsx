import {useForm, UseFormMethods} from "react-hook-form"
import {useDispatch} from "react-redux"
import {SampleCompanyCreation} from "shared/graphql/generated/globalTypes"
import {useCreateSampleCompany} from "shared/graphql/hooks"
import {navigateToRouteAction} from "../../../../redux/actions/navigation-action"
import {Route} from "../../../../routes"

export interface SampleCompanyCreationForm {
  readonly title: string
  readonly description: string
}

export interface UseCreateSampleCompanyHook {
  readonly createSampleCompany: (creation: SampleCompanyCreation) => void
  readonly createSampleCompanyLoading: boolean
  readonly formMethods: UseFormMethods<SampleCompanyCreationForm>
  readonly dismissModal: () => void
}

export const useCreateSampleCompanyModal = (): UseCreateSampleCompanyHook => {
  const dispatch = useDispatch()
  const {createSampleCompany, isCreateSampleCompanyLoading} = useCreateSampleCompany()
  const formMethods = useForm<SampleCompanyCreationForm>()

  const dismissModal = () => dispatch(navigateToRouteAction(Route.SampleCompanies))

  const handleCreate = (creation: SampleCompanyCreation) => {
    createSampleCompany(creation).then(response =>
      response.forEach(data => dispatch(navigateToRouteAction(Route.SampleCompanyDetail, {sampleCompanyId: data.id})))
    )
  }

  return {
    createSampleCompany: handleCreate,
    createSampleCompanyLoading: isCreateSampleCompanyLoading,
    formMethods,
    dismissModal
  }
}
