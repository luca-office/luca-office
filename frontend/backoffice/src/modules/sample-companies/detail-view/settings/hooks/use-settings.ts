import {useDispatch} from "react-redux"
import {SampleCompany} from "shared/models"
import {Option} from "shared/utils"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"

export interface UseSettingsHook {
  readonly navigateToDocuments: () => void
  readonly navigateToDomainSignature: () => void
  readonly navigateToErp: () => void
}

interface UseSettingsParams {
  readonly sampleCompany: Option<SampleCompany>
  readonly scenarioId?: UUID
}

export const useSettings = ({sampleCompany, scenarioId}: UseSettingsParams): UseSettingsHook => {
  const dispatch = useDispatch()

  const navigateToDocuments = () => {
    if (scenarioId) {
      sampleCompany.forEach(({id}) =>
        dispatch(navigateToRouteAction(Route.ScenarioAssignedSampleCompanyDocuments, {scenarioId, sampleCompanyId: id}))
      )
    } else {
      sampleCompany.forEach(({id}) =>
        dispatch(navigateToRouteAction(Route.SampleCompanyDocuments, {sampleCompanyId: id}))
      )
    }
  }

  const navigateToDomainSignature = () => {
    if (scenarioId) {
      sampleCompany.forEach(({id}) =>
        dispatch(
          navigateToRouteAction(Route.ScenarioAssignedSampleCompanyDomainSignature, {scenarioId, sampleCompanyId: id})
        )
      )
    } else {
      sampleCompany.forEach(({id}) =>
        dispatch(navigateToRouteAction(Route.SampleCompanyDomainSignature, {sampleCompanyId: id}))
      )
    }
  }

  const navigateToErp = () => {
    sampleCompany.forEach(({id}) =>
      !scenarioId
        ? dispatch(navigateToRouteAction(Route.SampleCompanyErp, {sampleCompanyId: id}))
        : dispatch(
            navigateToRouteAction(Route.ScenarioAssignedSampleCompanyDetailErp, {scenarioId, sampleCompanyId: id})
          )
    )
  }

  return {
    navigateToDocuments,
    navigateToDomainSignature,
    navigateToErp
  }
}
