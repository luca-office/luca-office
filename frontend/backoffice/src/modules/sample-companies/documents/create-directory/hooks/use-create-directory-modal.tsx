import {useForm, UseFormMethods} from "react-hook-form"
import {Directory} from "shared/models"
import {Option} from "shared/utils"
import {DirectoryCreation} from "../../../../../graphql/generated/globalTypes"
import {useCreateSampleCompanyDirectory} from "../../../../../graphql/hooks/mutations/directory/create-directory"
import {DirectoryCreationForm} from "../../../../common/files-and-directories/create-directory/create-directory-modal"

export interface UseCreateDirectoryHook {
  readonly createDirectory: (creation: DirectoryCreation) => Promise<Option<Directory>>
  readonly createDirectoryLoading: boolean
  readonly formMethods: UseFormMethods<DirectoryCreationForm>
}

export const useCreateDirectoryModal = (sampleCompanyId: UUID): UseCreateDirectoryHook => {
  const {createDirectory, createDirectoryLoading} = useCreateSampleCompanyDirectory(sampleCompanyId)
  const formMethods = useForm<DirectoryCreationForm>()

  return {
    createDirectory,
    createDirectoryLoading,
    formMethods
  }
}
