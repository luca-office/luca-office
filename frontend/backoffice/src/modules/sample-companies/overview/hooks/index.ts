export * from "./use-sample-companies"
