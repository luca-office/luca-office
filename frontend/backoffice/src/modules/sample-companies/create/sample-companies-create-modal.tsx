import {css} from "@emotion/react"
import * as React from "react"
import {Modal, Overlay, TextArea, TextInput} from "shared/components"
import {InputType} from "shared/enums"
import {spacingMedium, spacingSmall} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {SampleCompanyCreationForm, useCreateSampleCompanyModal} from "./hooks/use-create-sample-company-modal"

export const CreateSampleCompaniesModal: React.FC = () => {
  const {t} = useLucaTranslation()
  const {formMethods, dismissModal, createSampleCompany, createSampleCompanyLoading} = useCreateSampleCompanyModal()
  const {register, handleSubmit, errors} = formMethods

  const onSubmit = (formValues: SampleCompanyCreationForm) => {
    createSampleCompany({
      description: formValues.description,
      name: formValues.title,
      tags: []
    })
  }

  return (
    <Overlay>
      <Modal
        title={t("sample_companies__create_sample_company")}
        onDismiss={dismissModal}
        customStyles={styles.modal}
        onConfirm={handleSubmit(onSubmit)}
        preventSubmitOnEnter={true}
        confirmButtonDisabled={createSampleCompanyLoading}>
        <div css={styles.inputWrapper}>
          <TextInput
            ref={register({
              required: true
            })}
            placeholderKey="sample_companies__create_sample_company_title_placeholder"
            customStyles={styles.input}
            name="title"
            hasValidationError={!!errors?.title}
            type={InputType.text}
            labelKey={"title"}
          />
          <TextArea
            placeholder={t("sample_companies__create_sample_company_description_placeholder")}
            customStyles={[styles.input, styles.textarea]}
            ref={register({
              required: true
            })}
            name="description"
            rows={5}
            labelKey={"description"}
            hasValidationError={!!errors?.description}
          />
        </div>
      </Modal>
    </Overlay>
  )
}

const styles = {
  modal: css({width: 600}),
  input: css({
    width: "100%",
    marginBottom: spacingSmall
  }),
  textarea: css({
    resize: "none"
  }),
  inputWrapper: css({
    marginTop: spacingMedium
  })
}
