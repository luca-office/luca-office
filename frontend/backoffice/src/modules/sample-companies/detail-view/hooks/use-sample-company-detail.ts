import {isEmpty} from "lodash-es"
import {useForm, UseFormMethods} from "react-hook-form"
import {useDispatch} from "react-redux"
import {NotificationSeverity} from "shared/enums"
import {SampleCompanyUpdate} from "shared/graphql/generated/globalTypes"
import {SampleCompanyFragment} from "shared/graphql/generated/SampleCompanyFragment"
import {
  SampleCompanyProps,
  useDuplicateSampleCompany,
  usePublishSampleCompany,
  useSampleCompany,
  useUpdateSampleCompany
} from "shared/graphql/hooks"
import {AppNotification, SampleCompany} from "shared/models"
import {Option} from "shared/utils"
import {useCheckUserClaims} from "../../../../hooks/use-check-user-claims"
import {navigateToRouteAction} from "../../../../redux/actions/navigation-action"
import {updateNotification} from "../../../../redux/actions/ui/common-ui-action"
import {Route} from "../../../../routes"
import {NavigationButtonConfig} from "../../../../utils"

export interface SampleCompanyDetailForm {
  readonly name: string
  readonly description: string
  readonly usageField: string
  readonly audience: string
}

export interface UseSampleCompanyDetailHook extends Pick<SampleCompanyProps, "sampleCompany"> {
  readonly dataLoading: boolean
  readonly duplicateSampleCompany: () => void
  readonly duplicateSampleCompanyLoading: boolean
  readonly canBePublished: boolean
  readonly publishSampleCompany: (id: string) => void
  readonly publishSampleCompanyLoading: boolean
  readonly isPublished: boolean
  readonly formMethods: UseFormMethods<SampleCompanyDetailForm>
  readonly navigateToOverview: () => void
  readonly updateSampleCompany: (update: Partial<SampleCompanyUpdate>) => Promise<Option<SampleCompany>>
  readonly userMayArchive: boolean
}

export const useSampleCompanyDetail = (
  companyId: UUID,
  navigationButtonConfig?: NavigationButtonConfig
): UseSampleCompanyDetailHook => {
  const dispatch = useDispatch()
  const {sampleCompany, sampleCompanyLoading} = useSampleCompany(companyId)
  const {userClaims, userClaimsCheckLoading} = useCheckUserClaims(
    sampleCompany.map(sampleCompany => sampleCompany.authorId)
  )
  const {updateSampleCompany} = useUpdateSampleCompany()
  const {publishSampleCompany, isPublishSampleCompanyLoading} = usePublishSampleCompany()
  const {duplicateSampleCompany, duplicateSampleCompanyLoading} = useDuplicateSampleCompany()
  const formMethods = useForm<SampleCompanyDetailForm>()
  const navigateToOverview = () =>
    dispatch(
      navigateToRouteAction(navigationButtonConfig?.route ?? Route.SampleCompanies, navigationButtonConfig?.payload)
    )

  const handleUpdateSampleCompany = (update: Partial<SampleCompanyUpdate>) =>
    sampleCompany
      .map<Promise<Option<SampleCompany>>>(company => {
        const sampleCompanyUpdate: SampleCompanyUpdate = {
          description: update.description !== undefined ? update.description : company.description,
          name: update.name !== undefined ? update.name : company.name,
          tags: update.tags !== undefined ? update.tags : company.tags,
          emailSignature: update.emailSignature !== undefined ? update.emailSignature : company.emailSignature,
          domain: update.domain !== undefined ? update.domain : company.domain,
          logoFileId: update.logoFileId !== undefined ? update.logoFileId : company.logoFile?.id,
          profileFileId: update.profileFileId !== undefined ? update.profileFileId : company.profileFileId
        }
        return updateSampleCompany(companyId, sampleCompanyUpdate)
      })
      .getOrElse(Promise.resolve(Option.none<SampleCompanyFragment>()))

  const handleDuplicateSampleCompany = () => {
    duplicateSampleCompany(companyId)
      .then(duplicateOption =>
        duplicateOption.forEach(duplicate =>
          dispatch(navigateToRouteAction(Route.SampleCompanyDetail, {sampleCompanyId: duplicate.id}))
        )
      )
      .then(() =>
        dispatch(
          updateNotification(
            Option.of<AppNotification>({
              messageKey: "sample_companies__duplicate_success",
              severity: NotificationSeverity.Success
            })
          )
        )
      )
  }

  const isPublished = sampleCompany.exists(sampleCompany => sampleCompany.publishedAt !== null)
  const canBePublished = sampleCompany.exists(
    sampleCompany =>
      sampleCompany.profileFileId !== null && sampleCompany.logoFileId !== null && !isEmpty(sampleCompany.description)
  )

  return {
    dataLoading: sampleCompanyLoading,
    duplicateSampleCompany: handleDuplicateSampleCompany,
    duplicateSampleCompanyLoading: duplicateSampleCompanyLoading,
    canBePublished,
    publishSampleCompany,
    publishSampleCompanyLoading: isPublishSampleCompanyLoading,
    isPublished,
    formMethods,
    navigateToOverview,
    sampleCompany,
    updateSampleCompany: handleUpdateSampleCompany,
    userMayArchive: !userClaimsCheckLoading && userClaims.mayArchive
  }
}
