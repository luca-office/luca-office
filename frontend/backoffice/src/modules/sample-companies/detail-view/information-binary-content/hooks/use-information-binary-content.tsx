import {Dispatch, SetStateAction, useState} from "react"
import {FileUsageType, Relevance, SampleCompanyUpdate} from "shared/graphql/generated/globalTypes"
import {useCreateFile, useDeleteFileFromSampleCompany} from "shared/graphql/hooks"
import {SampleCompany, UploadBinary} from "shared/models"
import {useLucaTranslation, WithLucaTranslation} from "shared/translations"
import {first, Option, uploadBinary} from "shared/utils"

export interface UseInformationBinaryContentHook extends WithLucaTranslation {
  readonly showUploadIntroModal: boolean
  readonly showEditIntroModal: boolean
  readonly showViewIntroModal: boolean
  readonly setShowUploadIntroModal: Dispatch<SetStateAction<boolean>>
  readonly setShowEditIntroModal: Dispatch<SetStateAction<boolean>>
  readonly setShowViewIntroModal: Dispatch<SetStateAction<boolean>>
  readonly createLogo: (binaries: UploadBinary[]) => void
  readonly createIntro: (binaries: UploadBinary[]) => void
  readonly updateIntro: (file: File) => void
  readonly deleteLogo: (fileId: UUID) => void
  readonly deleteIntro: (fileId: UUID) => void
}

export const useInformationBinaryContent = (
  sampleCompanyOption: Option<SampleCompany>,
  updateSampleCompany: (update: Partial<SampleCompanyUpdate>) => Promise<Option<SampleCompany>>
): UseInformationBinaryContentHook => {
  const [showUploadIntroModal, setShowUploadIntroModal] = useState(false)
  const [showEditIntroModal, setShowEditIntroModal] = useState(false)
  const [showViewIntroModal, setShowViewIntroModal] = useState(false)
  const {t} = useLucaTranslation()
  const {createFile} = useCreateFile(sampleCompanyOption.map(({id}) => ({sampleCompanyId: id})).orUndefined())
  const {deleteEntity: deleteFileFromSampleCompany} = useDeleteFileFromSampleCompany(
    sampleCompanyOption.map(sampleCompany => sampleCompany.id).getOrElse("")
  )

  const handleCreateFile = (
    binaries: UploadBinary[],
    sampleCompanyUpdateFieldKey: keyof Pick<SampleCompanyUpdate, "profileFileId" | "logoFileId">
  ) => {
    first(binaries).map(binary =>
      sampleCompanyOption.forEach(sampleCompany =>
        createFile({
          binaryFileId: binary.data.id,
          name: binary.data.filename,
          relevance: Relevance.PotentiallyHelpful,
          tags: [],
          usageType: FileUsageType.FileSystem,
          directoryId: sampleCompany.directoryId
        }).then(fileOption =>
          fileOption.forEach(file => {
            updateSampleCompany({[sampleCompanyUpdateFieldKey]: file.id})
            setShowUploadIntroModal(false)
            setShowEditIntroModal(false)
          })
        )
      )
    )
  }

  const handleCreateLogo = (binaries: UploadBinary[]) => handleCreateFile(binaries, "logoFileId")
  const handleCreateIntro = (binaries: UploadBinary[]) => handleCreateFile(binaries, "profileFileId")

  const deleteLogo = (fileId: UUID) => {
    updateSampleCompany({logoFileId: null}).then(() => deleteFileFromSampleCompany(fileId))
  }

  const deleteIntro = (fileId: UUID) => {
    updateSampleCompany({profileFileId: null}).then(() => deleteFileFromSampleCompany(fileId))
  }

  const updateIntro = (file: File) => {
    uploadBinary(file).then(binary => handleCreateFile([binary], "profileFileId"))
  }
  return {
    createIntro: handleCreateIntro,
    createLogo: handleCreateLogo,
    deleteIntro,
    deleteLogo,
    setShowEditIntroModal,
    setShowUploadIntroModal,
    setShowViewIntroModal,
    showEditIntroModal,
    showUploadIntroModal,
    showViewIntroModal,
    t,
    updateIntro
  }
}
