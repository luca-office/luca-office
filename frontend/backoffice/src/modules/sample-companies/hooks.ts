import omit from "lodash-es/omit"
import * as React from "react"
import {SampleCompanyCreation} from "shared/graphql/generated/globalTypes"
import {useCreateSampleCompany} from "shared/graphql/hooks"
import {SampleCompany} from "shared/models"
import {Option, uploadBinary} from "shared/utils"

export interface CompanyCreation extends Omit<SampleCompanyCreation, "profileBinaryFileId" | "logoBinaryFileId"> {
  readonly logoFile: Option<File>
  readonly logoBase64: Option<string>
}

export interface CreateSampleCompanyProps {
  readonly createSampleCompany: (creation: CompanyCreation) => Promise<Option<SampleCompany>>
  readonly isCreateSampleCompanyLoading: boolean
}

export const useHandleSampleCompanyCreation = (): CreateSampleCompanyProps => {
  const [isCreateSampleCompanyLoading, setIsCreateSampleCompanyLoading] = React.useState<boolean>(false)

  const {createSampleCompany} = useCreateSampleCompany()
  return {
    createSampleCompany: (creation: CompanyCreation) =>
      new Promise<Option<SampleCompany>>((resolve, reject) => {
        if (creation.logoFile.isEmpty()) {
          reject(new Error("No logo file was provided."))
          return
        }
        setIsCreateSampleCompanyLoading(true)
        creation.logoFile.forEach(file =>
          uploadBinary(file)
            .then(binary => {
              createSampleCompany({
                ...omit(creation, ["logoFile", "logoBase64"])
              })
                .then(resolve)
                .catch(reject)
            })
            .catch(reject)
            .finally(() => setIsCreateSampleCompanyLoading(false))
        )
      }),
    isCreateSampleCompanyLoading
  }
}
