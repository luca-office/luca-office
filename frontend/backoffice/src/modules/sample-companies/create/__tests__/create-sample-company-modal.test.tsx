// importing from direct file because of issues of babel loader and spyOn
import {mount, shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Modal, Overlay, TextArea, TextInput} from "shared/components"
import {mockedFormMethods} from "sharedTests/utils/form-methods-mock"
import * as createSampleCompanyModal from "../hooks/use-create-sample-company-modal"
import {UseCreateSampleCompanyHook} from "../hooks/use-create-sample-company-modal"
import {CreateSampleCompaniesModal} from "../sample-companies-create-modal"

const hookValuesDefault: UseCreateSampleCompanyHook = {
  createSampleCompany: jest.fn(),
  createSampleCompanyLoading: false,
  formMethods: mockedFormMethods,
  dismissModal: jest.fn()
}

const stateSpy = jest.spyOn(createSampleCompanyModal, "useCreateSampleCompanyModal")

describe("create-sample-company-modal", () => {
  it("renders correctly", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)
    const component = create(<CreateSampleCompaniesModal />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault})

    const component = shallow(<CreateSampleCompaniesModal />)
    expect(component.find(Overlay)).toHaveLength(1)
    expect(component.find(Modal)).toHaveLength(1)
    expect(component.find(TextInput)).toHaveLength(1)
    expect(component.find(TextArea)).toHaveLength(1)
  })

  it("triggers dismiss correctly", async () => {
    const dismissModal = jest.fn()
    const createSampleCompany = jest.fn()
    const handleSubmit = jest.fn()
    stateSpy.mockReturnValue({
      ...hookValuesDefault,
      dismissModal,
      createSampleCompany: createSampleCompany,
      formMethods: {...mockedFormMethods, handleSubmit} as any
    })

    const component = mount(<CreateSampleCompaniesModal />)
    const onDismissHandler = component.find(Modal).prop("onDismiss")
    if (onDismissHandler) {
      onDismissHandler()
      expect(dismissModal).toHaveBeenCalledTimes(1)
    }
  })

  it("does not submit with field errors", async () => {
    const handleSubmit = jest.fn()
    const createSampleCompany = jest.fn()
    stateSpy.mockReturnValue({
      ...hookValuesDefault,
      createSampleCompany: createSampleCompany,
      formMethods: {
        ...mockedFormMethods,
        errors: {title: undefined, description: {type: "error", message: "Error at Field"}},
        handleSubmit
      } as any
    })

    const component = mount(<CreateSampleCompaniesModal />)

    component.find(".confirm-button").last().simulate("click")
    expect(handleSubmit).toHaveBeenCalledTimes(1)
    expect(createSampleCompany).toHaveBeenCalledTimes(0)
  })
})
