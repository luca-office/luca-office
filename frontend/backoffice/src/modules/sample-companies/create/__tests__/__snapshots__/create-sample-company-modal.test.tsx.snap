// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-sample-company-modal renders correctly 1`] = `
<div
  className="css-1iw6hgd-Overlay"
>
  <div
    className="css-1iw6hgd-Overlay"
  >
    <div>
      <div
        className="css-16c3j3g-Modal"
      >
        <header
          className="css-14sl779-Modal"
        >
          <div
            className="css-1f6p2vj-Heading"
          >
            sample_companies__create_sample_company
          </div>
          <div
            className="css-12vbzj8-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="css-1gs9x9w-Modal"
        >
          <div
            className="css-1rsfcj5-CreateSampleCompaniesModal"
          >
            <div
              className="css-jdxnts"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="css-13s2sfk-Heading"
              >
                title
              </div>
              <div
                className="css-1rjy8ih"
              >
                <input
                  className="css-pkbj54"
                  defaultValue=""
                  disabled={false}
                  name="title"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="sample_companies__create_sample_company_title_placeholder"
                  type="text"
                />
              </div>
            </div>
            <div
              className="css-192nlq7"
            >
              <div
                className="css-13s2sfk-Heading"
              >
                description
              </div>
              <textarea
                className="css-slakx0"
                name="description"
                placeholder="sample_companies__create_sample_company_description_placeholder"
                rows={5}
              />
            </div>
          </div>
        </section>
        <footer
          className="css-ukdqb3-Modal"
        >
          <button
            className="cancel-button css-a1i4d5-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              cancel_button
            </div>
          </button>
          <button
            className="confirm-button css-1ex7tyh-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              confirm_button
            </div>
          </button>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
